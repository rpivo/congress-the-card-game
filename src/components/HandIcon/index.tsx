import React from 'react';
import { Context } from '@components/App/store';
import { Actions } from '@utilities/types';
import Style from './style';

const HandIcon = (): JSX.Element => {
  const { dispatch, state } = React.useContext(Context);
  const { shouldShowHand } = state;

  const handleClick = (event: React.MouseEvent<HTMLInputElement>) => {
    event.stopPropagation();
    if (!shouldShowHand) dispatch({ type: Actions.SHOW_HAND });
  };

  return (
    <Style
      className='handIcon'
      onClick={(event: React.MouseEvent<HTMLInputElement>) => handleClick(event)}
      title='Hand'
    >
      <svg width='445.805' height='421.425' viewBox='0 0 445.805 421.425'>
        <g transform='translate(-195.558 -456.564)'>
          <g
            className='svgCard'
            transform='matrix(0.995, 0.105, -0.105, 0.995, 392.732, 495.977)'
            fill='#707070'
            stroke='#fff'
            strokeWidth='12'>
            <rect width='250' height='350' rx='15' stroke='none' />
            <rect x='6' y='6' width='238' height='338' rx='9' fill='none' />
          </g>
          <g
            className='svgCard'
            transform='translate(290.401 516.605) rotate(-9)'
            fill='#707070'
            stroke='#fff'
            strokeWidth='12'>
            <rect width='250' height='350' rx='15' stroke='none' />
            <rect x='6' y='6' width='238' height='338' rx='9' fill='none' />
          </g>
          <g
            className='svgCard'
            transform='matrix(0.914, -0.407, 0.407, 0.914, 195.558, 558.248)'
            fill='#707070'
            stroke='#fff'
            strokeWidth='12'>
            <rect width='250' height='350' rx='15' stroke='none' />
            <rect x='6' y='6' width='238' height='338' rx='9' fill='none' />
          </g>
        </g>
      </svg>
    </Style>
  );
};

export default HandIcon;
