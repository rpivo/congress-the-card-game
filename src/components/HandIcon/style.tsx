import styled from 'styled-components';

const Style = styled.div`
bottom: 25px;
cursor: pointer;
position: fixed;
right: 25px;  

  svg {
    height: 75px;
    width: 75px;

    &:hover .svgCard {
      fill: rgb(255, 255, 255);
      stroke: rgb(112, 112, 112);
    }
  }
`;

export default Style;
