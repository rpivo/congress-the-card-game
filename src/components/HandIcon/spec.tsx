import { mount } from 'enzyme';
import 'jest-styled-components';
import React from 'react';
import renderer from 'react-test-renderer';
import HandIcon from '@components/HandIcon';

describe('Deck', () => {

  it('should render correctly', () => {
    const tree = renderer
      .create(<HandIcon />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  describe('hover', () => {
    it('should update style on hover', () => {
      const wrapper = mount(<HandIcon />);
      expect(wrapper.find('div')).toHaveStyleRule(
        'fill',
        'rgb(255,255,255)',
        { modifier: 'svg:hover .svgCard' },
      );
      expect(wrapper.find('div')).toHaveStyleRule(
        'stroke',
        'rgb(112,112,112)',
        { modifier: 'svg:hover .svgCard' },
      );
    });
  });
});
