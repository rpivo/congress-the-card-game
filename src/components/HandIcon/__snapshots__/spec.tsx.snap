// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deck should render correctly 1`] = `
.c0 {
  bottom: 16px;
  cursor: pointer;
  position: fixed;
  right: 16px;
}

.c0 svg {
  height: 75px;
  width: 75px;
}

.c0 svg:hover .svgCard {
  fill: rgb(255,255,255);
  stroke: rgb(112,112,112);
}

@media screen and (max-width:1750px) {
  .c0 {
    bottom: 25px;
    right: 25px;
  }
}

<div
  className="c0 handIcon"
  onClick={[Function]}
  title="Hand"
>
  <svg
    height="421.425"
    viewBox="0 0 445.805 421.425"
    width="445.805"
  >
    <g
      transform="translate(-195.558 -456.564)"
    >
      <g
        className="svgCard"
        fill="#707070"
        stroke="#fff"
        strokeWidth="12"
        transform="matrix(0.995, 0.105, -0.105, 0.995, 392.732, 495.977)"
      >
        <rect
          height="350"
          rx="15"
          stroke="none"
          width="250"
        />
        <rect
          fill="none"
          height="338"
          rx="9"
          width="238"
          x="6"
          y="6"
        />
      </g>
      <g
        className="svgCard"
        fill="#707070"
        stroke="#fff"
        strokeWidth="12"
        transform="translate(290.401 516.605) rotate(-9)"
      >
        <rect
          height="350"
          rx="15"
          stroke="none"
          width="250"
        />
        <rect
          fill="none"
          height="338"
          rx="9"
          width="238"
          x="6"
          y="6"
        />
      </g>
      <g
        className="svgCard"
        fill="#707070"
        stroke="#fff"
        strokeWidth="12"
        transform="matrix(0.914, -0.407, 0.407, 0.914, 195.558, 558.248)"
      >
        <rect
          height="350"
          rx="15"
          stroke="none"
          width="250"
        />
        <rect
          fill="none"
          height="338"
          rx="9"
          width="238"
          x="6"
          y="6"
        />
      </g>
    </g>
  </svg>
</div>
`;
